ckan/lib/helpers.py

```diff
@@ -684,6 +685,10 @@ def are_there_flash_messages():
 
 def _link_active(kwargs):
     ''' creates classes for the link_to calls '''
+    if is_flask_request():
+        endpoint = request.url_rule.endpoint
+        action = request.url_rule.rule
+        return endpoint.split('.')[1] == action.translate('/')
```

Comment:
Maybe I'm missing something but this will never match a menu item (passed via `kwargs`) against the current endpoint loaded. I think it should work like this:

```diff
diff --git a/ckan/lib/helpers.py b/ckan/lib/helpers.py
index e451ba0..bc54a58 100644
--- a/ckan/lib/helpers.py
+++ b/ckan/lib/helpers.py
@@ -687,8 +687,11 @@ def _link_active(kwargs):
     ''' creates classes for the link_to calls '''
     if is_flask_request():
         endpoint = request.url_rule.endpoint
-        action = request.url_rule.rule
-        return endpoint.split('.')[1] == action.translate('/')
+        return (
+            kwargs.get('controller') == endpoint.split('.')[0] and
+            kwargs.get('action') == endpoint.split('.')[1]
+        )
+
     highlight_actions = kwargs.get('highlight_actions',
                                    kwargs.get('action', '')).split()
     return (c.controller == kwargs.get('controller')
```

Also to make easier to get rid of Pylons code in the future let's use separate functions for Flask / Pylons rather than doing checks inside. So something like:

```python

def _link_active(kwargs):
     ''' creates classes for the link_to calls '''
     if is_flask_request():
         return _link_active_flask(kwargs)
     else:
         return _link_active_pylons(kwargs)
```